src/test/java/pl/oskarkowalski/recruitment/ticketbookingapp/verification/TicketVerificationTest.java

```diff
@@ -0,0 +1,116 @@
+package pl.oskarkowalski.recruitment.ticketbookingapp.verification;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import pl.oskarkowalski.recruitment.ticketbookingapp.dtos.TicketDto;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Film;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Room;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Screening;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Seat;
+import pl.oskarkowalski.recruitment.ticketbookingapp.entities.Ticket;
+import pl.oskarkowalski.recruitment.ticketbookingapp.enums.SeatRow;
+import pl.oskarkowalski.recruitment.ticketbookingapp.enums.TicketPrice;
+import pl.oskarkowalski.recruitment.ticketbookingapp.repositories.ScreeningRepository;
+import pl.oskarkowalski.recruitment.ticketbookingapp.services.ReservationService;
+
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+public class TicketVerificationTest {
+
+  @InjectMocks
+  ReservationService reservationService;
+
+  @Mock
+  ScreeningRepository screeningRepository;
+
+  private final Film testFilm = new Film("Test", "Test", 200);
+  private final Room testRoom = new Room(100);
+
+  @Test
+  void testWeekendPrices() {
+    Clock clock = Clock.fixed(Instant.parse("2020-05-15T14:15:30.00Z"), ZoneId.of("CET"));
+    LocalDateTime fridayTime = LocalDateTime.now(clock);
+    LocalDateTime saturdayTime = LocalDateTime.now(clock).plusDays(1);
+    LocalDateTime sundayTime = LocalDateTime.now(clock).plusDays(2);
+
+    Screening testScreeningFriday = new Screening(testFilm, testRoom, fridayTime.plusMinutes(30));
+    Screening testScreeningSaturday = new Screening(testFilm, testRoom, saturdayTime.plusMinutes(30));
+    Screening testScreeningSunday = new Screening(testFilm, testRoom, sundayTime.plusMinutes(30));
+
+    assertThat(reservationService.checkTicketPrice(TicketPrice.ADULT, testScreeningFriday)).
```

Comment:
nie za dużo tych asercji? czy to nie powinny być osobne testy?

Reply:
W następnej wersji muszę zdekomponować te testy.